src/mono/wasm/debugger/BrowserDebugProxy/EvaluateExpression.cs

```diff
@@ -222,19 +213,37 @@ private object ConvertJSToCSharpType(JToken variable)
                 JToken value = variable["value"];
                 string type = variable["type"].Value<string>();
                 string subType = variable["subtype"]?.Value<string>();
-
                 switch (type)
                 {
                     case "string":
-                        return value?.Value<string>();
+                        return $"\"{value?.Value<string>()}\"";
                     case "number":
                         return value?.Value<double>();
                     case "boolean":
-                        return value?.Value<bool>();
+                        return value?.Value<string>().ToLower();
                     case "object":
-                        return variable;
+                    {
+                        if (subType == "null")
```

Comment:
is subtype ever anything but null here?

Reply:
Could also be an array, but in case of being an array it will have an objectId, so it's correct to run the code outside the if.

Reply:
so we don't need to pass the subtype through in that case?

Reply:
You are right, I've changed the code now :)
Thanks